### ISSUE TITLE
block_area contingent on road type

### ISSUE THREAD

**Issue by Guilherme-B** [graphhopper/graphhopper#1969]
The block_area feature is quite handy. However, blocking an entire area may have disastrous side effects - picture wanting to block a road which has an overpass/highway going on top of it, with the current model it means blocking the entire overpass section - possibly reflecting in the route adding quite a distance until the next overpass/highway junction.

A simple way to avoid this would be to add the possibility to add road/surface type exclusions per polygon as an optional parameter. In the previous case, it would mean adding the polygon with a highway exclusion, thus blocking the route on the road, but allowing it to function on the overpass.

**Comment by karussell** [comment]
Sorry, I'm not sure I understand the problem here. Can you rephrase this?

Combining blocking or avoiding an area with a road type exclusion will be possible via #1776 so we can probably close here (?)

**Comment by Guilherme-B** [comment]
@karussell absolutely, consider the following picture:

![image](https://user-images.githubusercontent.com/38859748/77648153-96c09080-6f5f-11ea-846b-64d4a5dc6c42.png)

Say you want to target the roads within the polygon to be under construction, thus avoidable by the routing engine - using block_area. However, the roads in yellow are under construction, but the overpass (in orange) isn't. Adding the polygon to the block_area with no proper road type discrimination, will mean the router will ignore the **whole** overpass from its route.

**Comment by karussell** [comment]
Ah, this is an interesting use case. Never thought about this. Thanks!

**Comment by easbar** [comment]
This is also useful for things like "avoid (too fast&dangerous) highways/Autobahn in Germany (but in other countries its ok" ?

**Comment by Guilherme-B** [comment]
@karussell my pleasure, let me know if you need help discussing the details

@easbar absolutely, I see it as fine-tuning what would otherwise be one of those methods which hides non desirable behavior
